### PR TITLE
Upgrade cargo from 0.40 to 0.41

### PR DESCRIPTION
**Pull request by emilk**
Requires a new dependency on cargo-platform.

### CODE REVIEW

File: cargo-geiger/src/cli.rs [hunk 1]
```diff
@@ -938,7 +942,8 @@ fn resolve_rs_file_deps(
         config: &config,
         spec: vec![],
         target: None,
-        release: false,
+        profile_specified: false,
+        profile_kind: ProfileKind::Release,
```
Comment (emilk) [inline cargo-geiger/src/cli.rs]: Is this correct?

**Comment by anderejd** [comment]
Thanks for the PR, I'll find some time soonish to look closer at it. 

We will also need to solve #86 first.

**Comment by emilk** [comment]
> We will also need to solve #86 first.

Can't help you with that I'm afraid.

**Comment by anderejd** [comment]
Inga problem!

It got solved by adding a new flag to rustup-init.exe.

**Comment by anderejd** [comment]
@emilk Please rebase on master and the CI should be working again.

File: cargo-geiger/src/cli.rs (hunk 1 above)
Comment (anderejd) [reply cargo-geiger/src/cli.rs]: The old flag was the boolean `release` set to false, the new one should be ProfileKind::Dev to match CompileMode::Check here:
https://github.com/anderejd/cargo-geiger/blob/cca096062a726a535ebfacefda328d58535b05e6/cargo-geiger/src/cli.rs#L672-L674

There's definitely a case for switching to Release profile as the default though, but let's do that in a new issue, unrelated to this library upgrade.

Comment (emilk) [reply cargo-geiger/src/cli.rs]: Of course, how silly of me! I have amended my commit to use `ProfileKind::Dev`.

**Comment by anderejd** [comment]
Thanks looks good!

**Comment by anderejd** [comment]
Released in 0.9.1.

**Pull request merged**